[Bug]: Problem with MetadataFilter and AstraDBVectorStore

### Bug Description

I have this code using LlamaIndex and AstraDB Vector Store

```
   astra_db_store = AstraDBVectorStore(
        token=token,
        api_endpoint=api_endpoint,
        collection_name=collection,
        embedding_dimension=1536
    )
    embed_model = .....
    llm = ....
    service_context = ....
    index = VectorStoreIndex.from_vector_store(
        vector_store=astra_db_store, service_context=service_context
    )

   
   filters = MetadataFilters(filters=[MetadataFilter(key="source", value="llama-index-ingest")])
   retriever = index.as_retriever(filters=filters)
```
 
I see this error:
```
  @staticmethod
    def _query_filters_to_dict(query_filters: MetadataFilters) -> Dict[str, Any]:
        if any(not isinstance(f, ExactMatchFilter) for f in query_filters.filters):
>           raise NotImplementedError("Only `ExactMatchFilter` filters are supported")
E           NotImplementedError: Only `ExactMatchFilter` filters are supported

../../../../../Library/Caches/pypoetry/virtualenvs/ragstack-ai-1_8bFyNQ-py3.11/lib/python3.11/site-packages/llama_index/vector_stores/astra.py:166: NotImplementedError
2023-12-11 12:04:47,472 [INFO] Test report line: test_astra.py::test_ingest_langchain_retrieve_llama_index -> ❌ <ExceptionInfo NotImplementedError('Only `ExactMatchFilter` filters are supported') tblen=22>   
```
   
Even if I change my code to using ExactMatchFilter

```
  filters = MetadataFilters(filters=[ExactMatchFilter(key="source", value="llama-index-ingest")])
  retriever = index.as_retriever(filters=filters)
```

The error is the same.
With the debugger I see that magically the MetadataFilters object contains a MetadataFilter filter object with the EQ operator.

### Version

0.9.11

### Steps to Reproduce

See the description above

### Relevant Logs/Tracbacks

_No response_

--- a/llama_index/vector_stores/astra.py
+++ b/llama_index/vector_stores/astra.py
@@ -11,11 +11,13 @@ from llama_index.indices.query.embedding_utils import get_top_k_mmr_embeddings
 from llama_index.schema import BaseNode, MetadataMode
 from llama_index.vector_stores.types import (
     ExactMatchFilter,
+    FilterOperator,
+    MetadataFilter,
     MetadataFilters,
     VectorStore,
     VectorStoreQuery,
     VectorStoreQueryMode,
-    VectorStoreQueryResult, FilterOperator, MetadataFilter,
+    VectorStoreQueryResult,
 )
 from llama_index.vector_stores.utils import (
     metadata_dict_to_node,
@@ -164,10 +166,15 @@ class AstraDBVectorStore(VectorStore):
     def _query_filters_to_dict(query_filters: MetadataFilters) -> Dict[str, Any]:
         # Allow only legacy ExactMatchFilter and MetadataFilter with FilterOperator.EQ
         if not all(
-                (isinstance(f, ExactMatchFilter) or
-                 (isinstance(f, MetadataFilter) and f.operator == FilterOperator.EQ))
-                for f in query_filters.filters):
-            raise NotImplementedError("Only filters with operator=FilterOperator.EQ are supported")
+            (
+                isinstance(f, ExactMatchFilter)
+                or (isinstance(f, MetadataFilter) and f.operator == FilterOperator.EQ)
+            )
+            for f in query_filters.filters
+        ):
+            raise NotImplementedError(
+                "Only filters with operator=FilterOperator.EQ are supported"
+            )
         return {f"metadata.{f.key}": f.value for f in query_filters.filters}
 
     def query(self, query: VectorStoreQuery, **kwargs: Any) -> VectorStoreQueryResult:

